crate/server/src/kms_server.rs

```diff
@@ -185,33 +185,35 @@ pub async fn prepare_kms_server(
     let google_cse_jwks = enable_google_cse.then_some(google_cse::list_jwks_uri());
 
     // Determine if JWT Auth should be used for authentication.
-    let (use_jwt_auth, jwt_config, jwks_manager) = if let Some(jwt_issuer_uri) =
-        &kms_server.params.jwt_issuer_uri
-    {
-        // Prepare all the needed URIs
-        let jwks_uri = JwtAuthConfig::uri(jwt_issuer_uri, kms_server.params.jwks_uri.as_deref());
-        let uris = google_cse_jwks.map_or_else(
-            || vec![jwks_uri.clone()],
-            |mut google_uris| {
-                google_uris.push(jwks_uri.clone());
-                google_uris
-            },
-        );
+    let (use_jwt_auth, jwt_config, jwks_manager) =
+        if let Some(jwt_config) = &kms_server.params.jwt_config {
+            // Prepare all the needed URIs from all the configured Identity Providers
+            let mut all_jwks_uris: Vec<_> = jwt_config
+                .iter()
+                .map(|idp_config| {
+                    JwtAuthConfig::uri(&idp_config.jwt_issuer_uri, Some(&idp_config.jwks_uri))
+                })
+                .collect();
+            // Add the one from google is cse is enabled
+            if let Some(google_cse_jwks) = google_cse_jwks {
+                all_jwks_uris.extend(google_cse_jwks)
+            }
```

Comment:
This comment seems misleading: you are not adding Google's ones if CSE is enable, but if they are Some, which may not be correlated to enabling CSE (it actually is, but I had to look some 12 lines above).
Suggested change
```suggestion
            if enable_google_cse {
                all_jwks_uris.extend(google_cse::list_jwks_uri())
            }
```

Also, this Google thing is spread into two different spots in the function: maybe you could come up with a way to merge them into one?

Reply:
good point - thks